test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/util/SmartRouterUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.integrationtests.util;
+
+import java.time.Duration;
+
+import org.kie.server.integrationtests.router.client.KieServerRouterClient;
+import org.kie.server.router.Configuration;
+
+public class SmartRouterUtils {
+
+    private static final Duration WAIT_STEP = Duration.ofSeconds(1);
+    private static final Duration MAX_WAIT_DURATION = Duration.ofSeconds(15);
+
+    public static void waitForContainerStart(KieServerRouterClient smartRouterAdminClient, String containerId) {
+        TimeUtils.wait(MAX_WAIT_DURATION, WAIT_STEP, () -> {
+            Configuration routerConfig = smartRouterAdminClient.getRouterConfig();
+            if (routerConfig.getContainerInfosPerContainer().containsKey(containerId)) {
```

Comment:
I'd throw this if-else statement away and simply
`return routerConfig.getContainerInfosPerContainer().containsKey(containerId)`

Reply:
Adjusted

Reply:
Thanks!